macros/src/approval/simple_multisig.rs

```diff
@@ -31,107 +31,37 @@ pub fn expand(meta: SimpleMultisigMeta) -> Result<TokenStream, darling::Error> {
                 ::near_contract_tools::slot::Slot::root(#storage_key)
             }
         }
 
-        use ::near_contract_tools::standard::nep141::Nep141;
-
-        #[::near_sdk::near_bindgen]
-        impl #imp ::near_contract_tools::standard::nep141::Nep141 for #ident #ty #wher {
-            #[payable]
-            fn ft_transfer(
-                &mut self,
-                receiver_id: ::near_sdk::AccountId,
-                amount: ::near_sdk::json_types::U128,
-                memo: Option<String>,
-            ) {
-                use ::near_contract_tools::{
-                    event::Event,
-                    standard::nep141::{Nep141Controller, Nep141Event},
-                };
-
-                ::near_sdk::assert_one_yocto();
-                let sender_id = ::near_sdk::env::predecessor_account_id();
-                let amount: u128 = amount.into();
-
-                let transfer = ::near_contract_tools::standard::nep141::Nep141Transfer {
-                    sender_id: sender_id.clone(),
-                    receiver_id: receiver_id.clone(),
-                    amount,
-                    memo: memo.clone(),
-                    msg: None,
-                };
-
-                #before_transfer
-
-                Nep141Controller::transfer(self, &sender_id, &receiver_id, amount, memo.as_deref());
-
-                #after_transfer
-            }
-
-            #[payable]
-            fn ft_transfer_call(
-                &mut self,
-                receiver_id: ::near_sdk::AccountId,
-                amount: ::near_sdk::json_types::U128,
-                memo: Option<String>,
-                msg: String,
-            ) -> ::near_sdk::Promise {
-                ::near_sdk::assert_one_yocto();
-                let sender_id = ::near_sdk::env::predecessor_account_id();
-                let amount: u128 = amount.into();
-
-                let transfer = ::near_contract_tools::standard::nep141::Nep141Transfer {
-                    sender_id: sender_id.clone(),
-                    receiver_id: receiver_id.clone(),
-                    amount,
-                    memo: memo.clone(),
-                    msg: None,
-                };
-
-                #before_transfer
-
-                let r = ::near_contract_tools::standard::nep141::Nep141Controller::transfer_call(
-                    self,
-                    sender_id.clone(),
-                    receiver_id.clone(),
-                    amount,
-                    memo.as_deref(),
-                    msg.clone(),
-                    near_sdk::env::prepaid_gas(),
-                );
-
-                #after_transfer
-
-                r
-            }
-
-            fn ft_total_supply(&self) -> ::near_sdk::json_types::U128 {
-                ::near_contract_tools::standard::nep141::Nep141Controller::total_supply(self).into()
-            }
+        // TODO: This pollutes the global namespace. Is there some better strategy?
```

Comment:
@encody Whom can you ask for advice about this?

Reply:
Unfortunately, probably not anybody at NEAR, since near-sdk-rs does the same thing.

Reply:
It could generate a module, but then that means that the users of the macro simply must be aware of the existence and name of the module instead of the enum (so it doesn't really solve the problem).

Reply:
Resolved in 45259d7